### PR TITLE
OTT-1338: Add support for video position targeting

### PR DESCRIPTION
**Pull request by Pubmatic-Dhruv-Sonone**
1. New VIDEO_POSITION_TYPE parameter in settings.py
2. Supported values for  **VIDEO_POSITION_TYPE:**  _PREROLL_, _MIDROLL_, _POSTROLL_, _ALL_
3. This parameter is supported for video and adpod setup
4. Optional parameter

Screenshot 2023-10-18 at 12 09 42 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,6 +132,7 @@ If successful, it returns all the orders in your GAM account.
 |`CURRENCY_EXCHANGE`|Same as **Currency Module** in the previous Line Item Tool. When used, this option converts the _rate_ calculated from CSV to the network's currency setting. This is applicable for `WEB`, `WEB_SAFEFRAME` and `NATIVE` only. |bool|True|
 |`OPENWRAP_USE_1x1_CREATIVE`| When this option is set, the tool creates a single creative with size 1x1 and all lineitems with size overrides. |bool|False|
 |`OPENWRAP_NATIVE_CREATIVE_USER_DEFINED_VAR`| When this option is set, the tool creates a string variable for NATIVE creative type. |string|None|
+|`VIDVIDEO_POSITION_TYPE`| determines which video position lineitem will target. Valid values -  `PREROLL`, `MIDROLL`, `POSTROLL`, `ALL`.This is a optional setting and is applicable for video and adpod setup. For adpod setup each slot lineitem will have same video position targeting |string|None|
```
Comment (pm-isha-bharti) [inline README.md:135]: Please rename the Config key.

File: dfp/create_line_items.py [hunk 2]
```diff
@@ -171,5 +171,8 @@ def create_line_item_config(name, order_id, placement_ids, ad_unit_ids, cpm_micr
     else:  # creative type is IN_APP_VIDEO
       line_item_config['videoMaxDuration'] = 15000
       line_item_config['targeting']['requestPlatformTargeting'] = {'targetedRequestPlatforms': ['MOBILE_APP','VIDEO_PLAYER']}
-    
+
+  if setup_type in ('Video', 'ADPOD'):
```
Comment (pm-isha-bharti) [inline dfp/create_line_items.py:175]: Please use existing constants

Comment (pm-isha-bharti) [reply dfp/create_line_items.py:175]: Do we want to have this setting for inapp_video and JW player as well? Please check

File: tests/test_add_new_openwrap_partner.py [hunk 3]
```diff
@@ -1209,7 +1209,7 @@ def test_setup_partner_for_RON(self, mock_get_users, mock_get_root_ad_unit_id,
                                                setup_type = constant.NATIVE, creative_template = None, num_creatives=2,
                                                use_1x1=False, currency_code='USD', custom_targeting= None,
                                                same_adv_exception= False, device_categories=None, device_capabilities=None, 
-                                               roadblock_type= 'ONE_OR_MORE', slot = None,adpod_creative_durations=None, creative_user_def_var = None, deal_config=None, deal_lineitem_enabled= False,  deal_config_type = None)
+                                               roadblock_type= 'ONE_OR_MORE', slot = None,adpod_creative_durations=None, creative_user_def_var = None, deal_config=None, deal_lineitem_enabled= False,  deal_config_type = None, video_position_type=None)
```
Comment (pm-isha-bharti) [inline tests/test_add_new_openwrap_partner.py:1212]: Also add test case for dfp.create_line_item_config

**Review by pm-isha-bharti — CHANGES_REQUESTED** [review]

File: docs/adpod_setup.md [hunk 4]
```diff
@@ -32,8 +32,8 @@
 |`DFP_SAME_ADV_EXCEPTION`|Determines whether to set the "Same Advertiser Exception" on line items. |bool|`False`|
 |`DFP_DEVICE_CATEGORIES`|Sets device category targetting for a Line item. Valid values: `Connected TV`, `Desktop`, `Feature Phone`, `Set Top Box`, `Smartphone`, and `Tablet`.|string or array of strings|None|
 |`DFP_ROADBLOCK_TYPE`| Roadblock Type. Set to  `ONE_OR_MORE` or `AS_MANY_AS_POSSIBLE`.|string|None|
-|`OPENWRAP_CUSTOM_TARGETING`|Array of extra targeting rules per line item. <br>  Ex:  `[("a", "IS", ("1", "2", "3")), ("b", "IS_NOT", ("4", "5", "6"))]` |array of arrays .)|None|
-
+|`OPENWRAP_CUSTOM_TARGETING`|Array of extra targeting rules per line item. <br>  Ex:  `[("a", "IS", ("1", "2", "3")), ("b", "IS_NOT", ("4", "5", "6"))]` |array of arrays|None|
+|`VIDVIDEO_POSITION_TYPE`| determines which video position lineitem will target. Valid values -  `PREROLL`, `MIDROLL`, `POSTROLL`, `ALL`.This is a optional setting and is applicable for video and adpod setup. For adpod setup each slot lineitem will have same video position targeting |string|None|
```
[the remark this answers is not in the document]
Comment (Pubmatic-Dhruv-Sonone) [reply docs/adpod_setup.md]: Done

File: dfp/create_line_items.py (hunk 2 above)
Comment (Pubmatic-Dhruv-Sonone) [reply dfp/create_line_items.py]: Done.  

Will check if we want to support this for other setups.

File: tests/test_add_new_openwrap_partner.py (hunk 3 above)
Comment (Pubmatic-Dhruv-Sonone) [reply tests/test_add_new_openwrap_partner.py:1212]: Done

File: README.md (hunk 1 above)
Comment (Pubmatic-Dhruv-Sonone) [reply README.md]: Done

File: dfp/create_line_items.py (hunk 2 above)
Comment (pm-isha-bharti) [reply dfp/create_line_items.py]: Please use constants defined in the constants file

Comment (pm-isha-bharti) [reply dfp/create_line_items.py]: Can we please track it on the parent ticket ?

**Pull request merged**